src/Makefile.am

```diff
@@ -519,13 +525,88 @@ EXTRA_DIST = tests
 # set the include path found by configure
 AM_CPPFLAGS = $(all_includes)
 
+suricata_SOURCES = suricata.c suricata.h $(COMMON_SOURCES)
+
 # the library search path.
 suricata_LDFLAGS = $(all_libraries) ${SECLDFLAGS}
-suricata_LDADD = $(HTP_LDADD) $(RUST_LDADD)
+suricata_LDADD = $(RUST_SURICATA_LIB) $(HTP_LDADD) $(RUST_LDADD)
+
+fuzz_applayerprotodetectgetproto_SOURCES = tests/fuzz/fuzz_applayerprotodetectgetproto.c $(COMMON_SOURCES)
```

Comment:
would it be possible to move those into a src/fuzz/Makefile.am or fuzz/Makefile.am? or perhaps a fuzz/applayerprotodetectgetproto/Makefile.am?

Reply:
I would love to.
The only way I see to achieve that is to compile `$(COMMON_SOURCES)` ie every source file except suricata which has `main` into a library (a convenience one).
I spent some hours trying and this is the best workaround I found, though it is not pretty.
The main problem is that C and Rust files depend on one another...

Any help from an autotool expert is welcome here :-)